File: two pointers/lc3.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    //base case
    if (s.length == 1){return 1}
    
    let sSet = new Set([])
    let lp = 0
    let size = 0


    for (let rp = 0; rp<s.length; rp++){
        while(sSet.has(s[rp])){
            sSet.delete(s[lp])
            lp++
        }
        
        sSet.add(s[rp])
        size = Math.max(rp-lp+1, size)
    }

    return size

};

File: two pointers/lc3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        
        if len(s) == 1:
            return 1
        
        lp = 0
        sSet = set()
        size = 0

        for rp in range(len(s)):
            while s[rp] in sSet:
                sSet.remove(s[lp])
                lp+=1
            
            sSet.add(s[rp])
            size = max(rp-lp+1, size)
        return size
        
